src/pixelator/plot/__init__.py

```diff
@@ -791,6 +791,129 @@ def plot_colocalization_diff_heatmap(
     return plt.gcf(), plt.gca()
 
 
+def _add_top_marker_labels(
+    differential_colocalization, n_top_pairs: int = 5, min_log_p: float = 5.0
+):
+    differential_colocalization = differential_colocalization.sort_values(
+        "median_difference"
+    )
+    differential_colocalization = differential_colocalization.loc[
+        -np.log10(differential_colocalization["p_adj"]) > min_log_p, :
+    ]
+
+    ## Labels for marker pair withs highest negative differential colocalization scores
+    for _, row in differential_colocalization.head(n_top_pairs).iterrows():
+        x, y = row[["median_difference", "p_adj"]]
+        y = -np.log10(y)
+        if x > 0:
+            continue
+        label = plt.text(
+            x, y, row["markers"], horizontalalignment="left", fontsize="xx-small"
+        )
+
+    ## Labels for marker pair with highest positive differential colocalization scores
+    for text, row in differential_colocalization.tail(n_top_pairs).iterrows():
+        x, y = row[["median_difference", "p_adj"]]
+        y = -np.log10(y)
+        if x < 0:
+            continue
+        label = plt.text(
+            x, y, row["markers"], horizontalalignment="left", fontsize="xx-small"
+        )
+
+
+def __add_target_mean_colocalizations(
+    differential_colocalization,
+    target_coloc,
+    value_col,
+):
+    differential_colocalization = differential_colocalization.fillna(0).reset_index()
+
+    target_values = (
+        target_coloc.groupby(["marker_1", "marker_2"])[value_col].mean().reset_index()
+    )
+    differential_colocalization = pd.merge(
+        target_values,
+        differential_colocalization,
+        on=["marker_1", "marker_2"],
+        how="inner",
+    )
+
+    return differential_colocalization
+
+
+def plot_colocalization_diff_volcano(
+    colocalization_data: pd.DataFrame,
+    target: str,
+    reference: str,
+    contrast_column: str = "sample",
+    cmap: str = "vlag",
+    use_z_score: bool = True,
+    n_top_pairs: int = 5,
+    min_log_p: float = 5.0,
+    ax: Optional[plt.Axes] = None,
+) -> Tuple[plt.Figure, plt.Axes]:
+    """Generate the volcano plot of differential colocalization between reference and target components.
+
+    Example usage: plot_colocalization_diff_volcano(pxl.colocalization, target:"stimulated", reference:"control", contrast_column="sample").
+
+    :param colocalization_data: The colocalization data frame that can be found in a pixel variable "pxl" through pxl.colocalization. The data frame should contain the columns "marker_1", "marker_2", "pearson", "pearson_z", and the contrast_column.
```

Comment:
```suggestion
    :param colocalization_data: The colocalization data frame that can be found in a pixel variable
                                                "pxl" through pxl.colocalization. The data frame should contain the 
                                                columns "marker_1", "marker_2", "pearson", "pearson_z", and the contrast_column.
```
Just breaking this up since the line is rather long. Unfortunately ruff does not enforce line lengths in doc strings, so we need to keep an eye out for this manually.